src/libpmempool/replica.c

```diff
@@ -561,30 +831,393 @@ check_badblocks(struct pool_set *set, struct poolset_health_status *set_hs)
 
 		for (unsigned p = 0; p < rep->nparts; ++p) {
 			const char *path = PART(rep, p)->path;
+			struct part_health_status *part_hs = &rep_hs->part[p];
 
 			int exists = os_access(path, F_OK) == 0;
 			if (!exists)
 				continue;
 
-			int ret = os_badblocks_check_file(path);
+			if (!part_hs->exist_recovery_file) {
+				/*
+				 * Bad blocks recovery file for this part
+				 * does not exist, so the process of saving
+				 * recovery files was interrupted.
+				 */
+				LOG(1,
+					"bad blocks recovery file for this part does not exist -- '%s'",
+					path);
+				return 1;
+			}
+
+			LOG(1,
+				"reading bad blocks from the recovery file -- '%s'",
+				part_hs->recovery_file);
+
+			ret = replica_part_badblocks_recovery_file_read(path,
```

Comment:
Maybe we should *first* check if all recovery files for parts exist and are valid, and then we can move on to reading those files one by one, knowing that it won't fail (unless something catastrophic happens, and we can just FATAL).
Right now you read in the blocks in one part, but the entire function can fail in later iterations. I'm not sure how this is handled, but it's an unnecessary error handling logic.

Reply:
Done